sktime/transformations/series/scaledlogit.py

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implmenents the scaled logit tranformation."""
+
+__author__ = ["ltsaprounis"]
+__all__ = ["ScaledLogitTransformer"]
+
+from copy import deepcopy
+
+import numpy as np
+
+from sktime.transformations.base import BaseTransformer
+
+
+class ScaledLogitTransformer(BaseTransformer):
+    r"""Scaled logit transform.
+
+    Combined with a sktime.forecasting.compose.TransformedTargetForecaster it ensures
+    that the forecast stays between the specified bounds (lower_bound, upper_bound).
+    If both lower_bound and upper_bound are not None, a scaled logit transform is
+    applied. Otherwise, it applies a log transform variation that ensures the
+    resulting forecast is bounded as specified. The transform is applied to all
+    scalar elements of the input array individually.
+
+    Parameters
+    ----------
+    lower_bound : float
+        lower bound
+    upper_bound : float
+        upper bound
+
+    See Also
+    --------
+    sktime.transformations.series.boxcox.LogTransformer :
+        Transformer input data using natural log. Can help normalize data and
+        compress variance of the series.
+    sktime.transformations.series.boxcox.BoxCoxTransformer :
+        Applies Box-Cox power transformation. Can help normalize data and
+        compress variance of the series.
+    sktime.transformations.series.exponent.ExponentTransformer :
+        Transform input data by raising it to an exponent. Can help compress
+        variance of series if a fractional exponent is supplied.
+    sktime.transformations.series.exponent.SqrtTransformer :
+        Transform input data by taking its square root. Can help compress
+        variance of input series.
+
+    Notes
+    -----
+    | The scaled logit transform is applied if both upper_bound and lower_bound are
+    | not None:
+    |   :math:`log(\frac{x - a}{b - x})`, where a is the lower and b is the upper bound.
+
+    | If upper_bound is None and lower_bound is not the transform applied is:
+    |   :math:`log(x - a)`
+
+    | If lower_bound is None and upper_bound is not the transform applied is:
```

Comment:
avoid elliptical references like "upper_bound is not", this is easy to misunderstand and easy to make more precise: "if upper_bound is not None" or "if upper_bound is passed"